enhancements/baremetal/baremetal-poison-pill-remediation.md

```diff
@@ -0,0 +1,189 @@
+---
+title: baremetal-poison-pill-remediation
+authors:
+  - "@n1r1"
+  - "@abeekhof"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-11-15
+last-updated: 2020-11-15
+status: implementable
+see-also:
+  - https://github.com/openshift/enhancements/blob/master/enhancements/machine-api/machine-health-checking.md 
+replaces:
+  - None
+superseded-by:
+  - None
+---
+
+# Baremetal Poison Pill Remediation
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Existing baremetal remediation strategies utilize BMC credentials to power-cycle and/or reprovision the host.
+However there are also environments that either do not include BMCs, or there are policies in place that prevent
+them from being utilized. Such environments would also benefit from the ability to safely recover affected workloads
+and restore cluster capacity (where possible).
+
+This proposal describes an alternate mechanism for a node in a cluster
+to detect its health status and take actions to remediate itself in case of a failure. While not all remediation events
+can result in the node returning to a healthy state, the proposal does allow surviving parts of the cluster to assume
+the node has reached a safe state so that its workloads can be automatically recovered.
+
+
+This work can also be useful for clusters **with** BMC credentials.
+If there’s a network outage between the node running CAPBM (which expects to use BMC commands like power-off etc.) and the
+unhealthy node, the power-cycle attempt cannot
+succeed. Self health checks and remediation can resolve such cases, however that work/scenario is out of scope for
```

Comment:
Using poison pill in BMC clusters is possible as escalation path (e.g. first try to reboot with BMC, if it didn't work poison-pill kicks in). 
To simplify the proposal we wanted to avoid outlining how this escalation path can work and how this coordination should be done.
We mention the benefit for BMC clusters just as a something we might want to do in the future, or ways to expand that feature.